### ISSUE TITLE
Matrix blocks are soft-deleted

### ISSUE THREAD

**Issue by darylknight** [craftcms/cms#3846]
### Description

I have a site where each entry has a matrix field, and in that matrix field there's an Entries field which sets relations. Today the client deleted a matrix block, but the target element was still showing up on the page as being related. I ran garbage collection and the element disappeared. I'm therefore pretty sure that it was being caused by the matrix block getting soft-deleted. I don't know whether it's deliberate that matrix blocks are soft-deleted, but there's nowhere to see deleted matrix blocks to permanently remove them (and it's another step for the client.).

Is this a bug with soft-deletion, or a UI problem?

### Steps to reproduce

1. Set up a matrix block with an entries field and output the relations
2. Delete the matrix block
3. Relation still exists.

### Additional info

- Craft version: 3.1.11
- PHP version: 7.2
- Database driver & version: MySQL

**Comment by brandonkelly** [comment]
Yep, the `relatedTo` param wasn’t taking soft-deleted Matrix blocks into account. Fixed for the next release.

If you need to get this fix right away, open `composer.json` and change your `craftcms/cms` requirement:

```json
"require": {
  "craftcms/cms": "dev-develop#cb35dbbb03db63b0c49b35f436c7d1ce44c2ecc0 as 3.1.11",
  "...": "..."
}
```

Then run `composer update` from your terminal.

**Comment by brandonkelly** [comment]
We just released 3.1.12 with the fix.

**Comment by darylknight** [comment]
Thank you!
